db/scripts/3_create_l2_tables.sql

```diff
@@ -0,0 +1,51 @@
+\connect rollup;
+
+
+CREATE TABLE optimistic_canonical_chain_batch (
+  id BIGSERIAL NOT NULL,
+  batch_number BIGINT NOT NULL,
+  submitted_tx_batch_l1_tx_hash character(66) DEFAULT NULL,
+  submitted_root_batch_l1_tx_hash character(66) DEFAULT NULL,
+  tx_batch_status TEXT NOT NULL DEFAULT 'QUEUED',
+  root_batch_status TEXT NOT NULL DEFAULT 'QUEUED',
+  created TIMESTAMP NOT NULL DEFAULT NOW(),
+  PRIMARY KEY (id),
+  UNIQUE (batch_number)
+);
+CREATE INDEX optimistic_canonical_chain_batch_tx_status_idx ON optimistic_canonical_chain_batch USING btree (tx_batch_status);
+CREATE INDEX optimistic_canonical_chain_batch_root_status_idx ON optimistic_canonical_chain_batch USING btree (root_batch_status);
+
+CREATE TABLE l2_tx_output (
+  id BIGSERIAL NOT NULL,
+  l1_rollup_tx_id BIGINT DEFAULT NULL,
+  occ_batch_number BIGINT DEFAULT NULL,
+  occ_batch_index INT DEFAULT NULL,
```

Comment:
Assuming `occ` is `optimistic_canonical_chain`, this will presumably only be used by the sequencer.  In that case, are there any equivalent entries needed for the verifier?  Maybe not, and the upstream tables are sufficient to recover with `l1_rollup_tx_id`, but thought I'd ask.

Reply:
Yeah, this is where the naming is tough. Really, it's the same. Like everything else, the _optimistic_ batch becomes _the batch_ on L1. If we name it `batch_number` / `batch_index`, that's ambiguous, but if we name it `occ_batch_number` / `occ_batch_index` that surfaces this question. Can't really win.

Separate from naming, this brings up a good point: the Verifier will still need to run the `OptiimsticCanonicalChainBatchCreator` for this exact reason (see: [how the verifier gets its data](https://github.com/ethereum-optimism/optimism-monorepo/pull/187/files/6c07839a68bff753eddd65628b26bfedafb59241#diff-1fbd60a05b299724ab653f176d725dbdR35)), but *not* the `OptimisticCanonicalChainBatchSubmitter`. That could be reason to rename the rename 🤦

Reply:
For now, I'll leave it, but if you have ideas, I'm open 👍

Reply:
FWIW, named it `batch_number` and renamed `OptimisticCanonicalChainBatch` to `CanonicalChainBatch`. The entries in our table should always match or be _slightly_ ahead of the L1 canonical chain. In the event that our entry that is ahead of the L1 canonical chain somehow doesn't end up matching the L1 canonical chain, it will be deleted and updated to match the L1 chain, so this naming should be consistent with what it represents.